crates/core_arch/src/aarch64/neon/mod.rs

```diff
@@ -1799,6 +1799,14 @@ pub unsafe fn vqtbx4q_p8(a: poly8x16_t, t: poly8x16x4_t, idx: uint8x16_t) -> pol
     ))
 }
 
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(ldr))]
+pub unsafe fn vld1q_f32(addr: *const f32) -> float32x4_t {
+    use crate::core_arch::simd::f32x4;
+    transmute(f32x4::new(*addr, *addr.offset(1), *addr.offset(2), *addr.offset(3)))
+}
```

Comment:
It would be better to have a single implementation of this in `arm/neon` which is shared by both aarch64 and arm.

Reply:
I agree but went with separate implementations because clang's implementations were separate:

arm:
https://github.com/llvm/llvm-project/blob/d5c45041f1465f4ecc3828efbbb27aa7b4d23d89/clang/lib/CodeGen/CGBuiltin.cpp#L5875

aarch64:
https://github.com/llvm/llvm-project/blob/d5c45041f1465f4ecc3828efbbb27aa7b4d23d89/clang/lib/CodeGen/CGBuiltin.cpp#L10607

I'm happy to unify them if you think that's better.

Reply:
Ah good point! Indeed ARM's case is different since the `VLD1` intrinsic takes additional alignment information as an operand. In this case it's fine to leave it separate.